### ISSUE TITLE
AbstractYouTubePlayerListener onReady() not called sometimes- frequency 4/10 

### ISSUE THREAD

**Issue by ambedkar0847** [PierfrancescoSoffritti/android-youtube-player#643]
XML part : 
----------------------------------------------------------------
  <com.pierfrancescosoffritti.androidyoutubeplayer.core.player.views.YouTubePlayerView
                        android:id="@+id/youTubePlayerView"
                        android:layout_width="match_parent"
                        android:layout_height="wrap_content"
                        android:visibility="gone"
                        app:autoPlay="false"
                        app:handleNetworkEvents="false"
                        app:showFullScreenButton="false"
                        app:showYouTubeButton="false" />

------------------------------------------------------------------------------
Java part : 

  var videoId  = "xXIhxNYiWLI"
                        youTubePlayerView?.addYouTubePlayerListener(object : AbstractYouTubePlayerListener() {
                            override fun onReady(youTubePlayer: YouTubePlayer) {
                                youTubePlayer.seekTo(1f)

                                if (!videoId.isNullOrBlank()) {
                                    youTubePlayer.cueVideo(videoId, 0f)
                                } else {
                                    Toast.makeText(context,"Error in playing video",Toast.LENGTH_SHORT).show()
                                }
                            }
                        })

I am calling above logic in retrofit response method.

Can you please help on what circumstances the app will not get "onReady" callback from library.

Thanks for your help.

**Comment by malliaridis** [comment]
Hello @ambedkar0847. Have you checked if the video id you get is `null` and therefore the `youTubePlayer.cueVideo()` method is not called at all? 

And can you please use \`\`\`xml ... \`\`\` and \`\`\`java ... \`\`\` for correct formatting in your code? It will get easier to read and to copy for reproduction.

**Comment by ambedkar0847** [comment]
@malliaridis 
Yes, I checked video is not null, and please find formatted code snippet 

 var videoId  = "xXIhxNYiWLI"
 youTubePlayerView?.addYouTubePlayerListener(object : AbstractYouTubePlayerListener() {
 override fun onReady(youTubePlayer: YouTubePlayer) {
 youTubePlayer.seekTo(1f)
 if (!videoId.isNullOrBlank()) {
     youTubePlayer.cueVideo(videoId, 0f)
    } else {
             Toast.makeText(context,"Error in playing video",Toast.LENGTH_SHORT).show()
           }
       }
    })

**Comment by NiazSagor** [comment]
Having the same issue and a bit strange also.
I am using the player in 2 places. One in _view_ inside view pager and another one is in _activity_. The activity one works fine. But the other one does not work. onReady is never called

**Comment by braver-tool** [comment]
I am also facing the same issue. And am using the video view in 3 places. One is Fragment, second one is PIP activity and the last one is normal activity. PIP activity is working all time.But other two is not working. onReady interface never fired.

**Comment by HassanMohammed1997** [comment]
look at this @braver-tool
https://github.com/PierfrancescoSoffritti/android-youtube-player/issues/685

**Comment by Wayne-Wang-Wayne** [comment]
I'm facing the same issue, is there a fix?
